seungjong/services/userService.js

```diff
@@ -1,19 +1,44 @@
 // services/userService.js
 const userDao = require("../models/userDao");
+const bcrypt = require("bcrypt");
 
-const { passwordValidationCheck } = require("../utils/validationCheck");
+const jwt = require("jsonwebtoken");
+
+const secretKey = process.env.SECRETKEY;
 
 const signUp = async (name, email, password, profileImg) => {
-  await passwordValidationCheck(password);
+  const hashedPassword = await bcrypt.hash(password, 12);
+  console.log(`1111111111`, hashedPassword);
 
   const createUser = await userDao.createUser(
     name,
     email,
-    password,
+    hashedPassword,
     profileImg
   );
 
   return createUser;
 };
 
-module.exports = { signUp };
+const checkHash = async (password, hashedPassword) => {
+  return await bcrypt.compare(password, hashedPassword);
+};
+
+const logIn = async (email, password) => {
+  const [login] = await userDao.logIn(email);
+  const hashed = login.password;
+  const result = await checkHash(password, hashed);
+
+  if (!result) {
+    const error = new Error("Invalid User");
+    error.statusCode = 400;
+    throw error;
+  }
+  const payLoad = { userId: login.id };
+  return jwt.sign(payLoad, secretKey);
+};
```

Comment:
```suggestion
const logIn = async (email, password) => {
  const [user] = await userDao.getUserByEmail(email);
  
  if (!user || !bcrypt.compare(password, user.password)) throw new Error("Invalid Email or Password");
 
  return jwt.sign({ userId: user.id }, process.env.SECRETKEY);
};
```
suggestion을 참고해서 리팩토링 해주세요!

dao는 Data Access Object의 약자로서 해당 부분에 작성되는 함수이름으로 login은 적합하지 않습니다. CRUD(create, get, update, delete)와 자원의 조합으로 getUserByEmail과 같은 식으로 작성해주세요!

또한 (payload, secretKey) 같이 가독성을 해치지않는 선에서 반복되어 사용되는 변수가 아니라면, 메모리 소비 없이 곧바로 필요한 부분에 작성될 수 있게 해주세요.